katdal/sensordata.py

```diff
@@ -462,6 +462,73 @@ def _safe_linear_interp(xi, yi, x):
     return (1.0 - end_weight) * yi[start] + end_weight * yi[end]
 
 
+_KATCP_DECODER = {'integer': int, 'float': float, 'boolean': lambda x: x == '1',
+                  'lru': str, 'timestamp': float, 'discrete': str,
+                  'address': str, 'string': str}
+
+
+def get_sensor_from_katstore(store, name, start_time, end_time):
+    """Get raw sensor data from katstore (CAM's central sensor database).
+
+    Parameters
+    ----------
+    store : string
+        Hostname / endpoint of katstore webserver speaking katstore REST API
+    name : string
+        Sensor name (the escaped version with underscores)
+    start_time, end_time : float
+        Time range for sensor records as UTC seconds since Unix epoch
+
+    Returns
+    -------
+    data : :class:`RecordSensorData` object
+        Retrieved sensor data with 'timestamp', 'value' and 'status' fields
+
+    Raises
+    ------
+    ConnectionError
+        If this cannot connect to the katstore server
+    ValueError
+        If connection succeeded but interaction with REST API failed
+    KeyError
+        If the sensor was not found in the store or it has no data
+    """
+    # First check existence of sensor and get its KATCP type
+    url = "http://{}/katstore/sensors".format(store)
+    try:
+        result = requests.get(url, params={'sensors': name})
+    except requests.exceptions.ConnectionError as exc:
+        raise_from(ConnectionError("Could not connect to sensor store '%s'" %
+                                   (store,)), exc)
+    try:
+        sensor_info = {rec[0]: rec[2] for rec in result.json()}
+    except ValueError as exc:
```

Comment:
There are probably other errors that could occur e.g. IndexError if the record doesn't have 3 entries, and TypeError if it isn't a list of lists.

Reply:
Good, I'll catch those too.